core/string/ustring.h

```diff
@@ -523,11 +523,19 @@ class String {
 	CharString ascii(bool p_allow_extended = false) const;
 	CharString utf8() const;
 	Error parse_utf8(const char *p_utf8, int p_len = -1, bool p_skip_cr = false);
-	static String utf8(const char *p_utf8, int p_len = -1);
```

Comment:
maybe `_FORCE_INLINE_`, even if the compiler will probably do it either way

Reply:
That could be useful sometime. For example, debug build will not inline them (debug build do not inline anything by default), but if your debug build is too slow, and what you want to debug is not in an inline function, you can force all those _FORCE_INLINE_ to inline even in debug with `/d2Obforceinline` or `/Ob1` (in MSVC) to regain some speed.

Reply:
That's a good point actually. Making non-optimizing builds bearable in speed is a worthwhile endeavour.

Reply:
I'm also not a big fan of `_FORCE_INLINE_`, I think Godot overuses it in general and it should be more targeted, there is little considerations of the downsides.

But I did do a test a few years ago and runtime performance was admittedly a tad faster with it overall (versus being macrod out).

Note it's compiled out in `DEV` builds.